specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json

```diff
@@ -8127,13 +8502,6 @@
             "type": "string"
           }
         },
-        "roleAssignments": {
```

Comment:
why is this property being removed?

Reply:
Om Nishant <omnishan@microsoft.com>: After the roles and responsibilities were revisited to keep up with compliance and customer expectations, extraneous fields were removed to prevent any misleading and unrealistic expectations.

Reply:
removing properties from an existing API-versions is a breaking change. It should be done in a new api-version

Reply:
what happens when customer sets this property using an old api-version which supported it and gets the resource using new api-version. And then may be after that, does a PUT again from the new api-version

Reply:
roleAssignments under AzureToAzureVmSyncedConfigDetails is present only in response of GET ReplicationProtectedItem.

Actually these are role assignments of the Virtual Machine that is getting replicated (Virtual Machine and ReplicationProtectedItem have 1:1 association in ASR).
We were storing & syncing roleAssignments of the Virtual Machine - which we have stopped doing (shared the reason from Om Nishant in previous comment).

So, there is no PUT and only GET associated with the property which we have removed.